velox/exec/ProbeOperatorState.h

```diff
@@ -36,11 +36,17 @@ enum class ProbeOperatorState {
   kWaitForBuild = 0,
   /// The running state that join the probe input with the build table.
   kRunning = 1,
-  /// Wait for all the peer probe operators to finish processing inputs. This
-  /// state only applies when disk spilling is enabled. The last finished
-  /// operator will notify the build operators to build the next hash table from
-  /// the spilled data. Then all the peer probe operators will wait for the next
-  /// hash table to build.
+  /// This state has different handlings for hash and nested loop join probe.
+  /// For hash probe, wait for all the peer probe operators to finish processing
+  /// inputs.
+  /// This state only applies when disk spilling is enabled. The last finished
```

Comment:
Fix comment:
"This state only applies when disk spilling is enabled." not true anymore.
Number the usecases for this state.
1. For hashprobe, wait for peer probe operator...
2. Disk spilled case....